src/Storage/StreamWrapper.php

```diff
@@ -0,0 +1,612 @@
+     */
+    public function dir_rewinddir()
+    {
+        $this->directoryGenerator = $this->bucket->objects([
+            'prefix' => $this->file,
+            'fields' => 'items/name,nextPageToken'
+        ]);
+        return true;
+    }
+
+    /**
+     * Callback handler for trying to create a directory.
+     *
+     * @param string $path The url directory to creaet
+     * @param int $mode The permissions on the directory
+     * @param int $options Bitwise mask of options
+     * @return bool
+     */
+    public function mkdir($path, $mode, $options)
+    {
+        $client = $this->openPath($path);
+        $this->file = $this->makeDirectory($this->file);
+
+        try {
+            $this->bucket->upload('', [
+                'name' => $this->file
+            ]);
+        } catch (ServiceException $e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Callback handler for trying to move a file or directory.
+     *
+     * @param string $from The URL to the current file
+     * @param string $to The URL of the new file location
+     * @return bool
+     */
+    public function rename($from, $to)
+    {
+        $url = parse_url($to);
+        $destinationBucket = $url['host'];
+        $destinationPath = substr($url['path'], 1);
+
+        $this->dir_opendir($from, []);
+        foreach ($this->directoryGenerator as $file) {
+            $name = $file->name();
+            $newPath = str_replace($this->file, $destinationPath, $name);
+
+            $obj = $this->bucket->object($name);
+            $obj->rename($newPath, ['destinationBucket' => $destinationBucket]);
+        }
+        return true;
+    }
+
+    /**
+     * Callback handler for trying to remove a directory..
+     *
+     * @param string $path The URL directory to remove
+     * @param int $options Bitwise mask of options
+     * @return bool
+     */
+    public function rmdir($path, $options)
+    {
+        $path = $this->makeDirectory($path);
+        return $this->unlink($path);
+    }
+
+    /**
+     * Callback handler for retrieving the underlaying resource
+     *
+     * @param int $castAs STREAM_CAST_FOR_SELECT|STREAM_CAST_AS_STREAM
+     * @return resource
+     */
+    public function stream_cast($castAs)
+    {
+        return false;
+    }
+
+    /**
+     * Callback handler for deleting a file
+     *
+     * @param string $path The URL of the file to delete
+     * @return bool
+     */
+    public function unlink($path)
+    {
+        $client = $this->openPath($path);
+        $object = $this->bucket->object($this->file);
+
+        try {
+            $object->delete();
+            return true;
+        } catch (ServiceException $e) {
+            return false;
+        }
+    }
+
+    /**
+     * Callback handler for retrieving information about a file
+     *
+     * @param string $path The URI to the file
+     * @param int $flags Bitwise mask of options
+     * @return array|false
```

Comment:
The return should document a type, not a value. I would suggest `array|bool`. This comes up in a few of the other methods in this class.